java/bulk-export/bulk-export-lambda/src/main/java/sleeper/bulkexport/lambda/SqsBulkExportProcessorLambda.java

```
@@ -0,0 +1,146 @@
+     * @param context The AWS Lambda context object providing runtime information.
+     * @return Always returns null.
+     */
+    @Override
+    public Void handleRequest(SQSEvent event, Context context) {
+        event.getRecords().stream()
+                .map(SQSEvent.SQSMessage::getBody)
+                .peek(body -> LOGGER.debug("Received message with body {}", body))
+                .flatMap(body -> deserialiseAndValidate(body).stream())
+                .peek(json -> LOGGER.debug(json.toString()))
+                .forEach(query -> {
+                    try {
+                        processor.processExport(query);
+                    } catch (ObjectFactoryException e) {
+                        LOGGER.error("Failed to process export query", e);
+                    }
+                });
+        return null;
+    }
+
+    /**
+     * Deserialises a JSON string into a BulkExportQuery object and validates it.
+     * If the deserialization or validation fails, logs the error and returns an
+     * empty Optional.
+     *
+     * @param message the JSON string representing the BulkExportQuery
+     * @return an Optional containing the deserialized BulkExportQuery if
+     *         successful, otherwise an empty Optional
+     */
+    public Optional<BulkExportQuery> deserialiseAndValidate(String message) {
+        try {
+            BulkExportQuery exportQuery = bulkExportQuerySerDe.fromJson(message);
+            LOGGER.debug("Deserialised message to export query {}", exportQuery);
+            return Optional.of(exportQuery);
+        } catch (BulkExportQueryValidationException e) {
+            LOGGER.error("QueryValidationException validating export query from JSON {}", message, e);
+            return Optional.empty();
+        } catch (RuntimeException e) {
+            LOGGER.error("Failed deserialising export query from JSON {}", message, e);
+            return Optional.empty();
```

I'm not sure we want to discard these failures? We could throw the exceptions all the way out, or report just the ones that failed with SQSBatchResponse. Either way would retry the messages or send them to the dead letter queue, whereas as it is these failures are just ignored.

---

Thinking about it, it makes sense not to retry validation failures, but there's no way for the client to tell it failed. Maybe we should send these directly to the dead letter queue?

This is different from the similar behaviour in an ingest task, because there the failures are reported to the job tracker, although there's an argument those should go to the dead letter queue too.

---

This will be handled in https://github.com/gchq/sleeper/issues/4334